main.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  def my_each
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    result = []
+    my_each { |e| result.push e if yield e }
+    result
+  end
+
+  def my_all?
+    result = true
+    my_each do |e|
+      result = false unless yield e
+    end
+    result
+  end
+
+  def my_any?
+    result = false
+    my_each do |e|
+      if yield e
+        result = true
+        break
+      end
+    end
+    result
+  end
+
+  def my_none?
+    result = true
+    my_each do |e|
+      if yield e
+        result = false
+        break
+      end
+    end
+    result
+  end
+
+  def my_count(num = nil)
+    count = 0
+    if num
+      my_each do |e|
+        count += 1 if e == num
+      end
+    else
+      my_each do |e|
+        count += 1 if yield e
+      end
+    end
+    count
+  end
+
+  def my_map(proc = nil)
+    result = []
+    if proc
+      my_each_with_index do |e, i|
+        result[i] = proc.call(e)
+      end
+    else
+      my_each_with_index do |e, i|
+        result[i] = yield e
+      end
+    end
+    result
+  end
+
+  def my_inject(*args)
```

Comment:
Your method  returns this error
```
  NoMethodError:
        undefined method `nextif'
```
**Caused by** 
```
test= proc  {|a,b| a + b}
array.my_inject(&test) == array.inject(&test)
```

Reply:
thanks for the review. I fixed them all